feat: 미니 플레이어 클릭 시 채널 플레이어로 이동 기능 구현

devjaeyoon opened the pull request:
### ✨ 이슈 번호

- #118 

### 📌 설명

미니 플레이어 클릭 시 채널 플레이어로 이동하는 기능을 구현한 Pull Request입니다.

### 📃 작업 사항

- [x] 미니 플레이어 클릭 시 채널 플레이어로 이동

### 💭 리뷰 사항 반영

- [x] 중복 클릭 방지 함수명 `preventDuplicateClick`으로 변경

### ✅ Pull Request 체크 사항

- [x] 가장 최신 브랜치를 pull 하였습니다.
- [x] base 브랜치명을 확인하였습니다.
- [x] 코드 컨벤션을 모두 확인하였습니다.
- [x] 브랜치명을 확인하였습니다.

devjaeyoon committed:
feat: 미니 플레이어 클릭 시 채널 플레이어로 이동하는 기능 추가

- MiniPlayer 클릭 시 현재 재생 중인 채널의 채널 플레이어로 라우팅되도록 처리
- useChannelNavigation 훅 생성 및 적용
- 채널 전환 시 selectedChannelId와 isChannelChanged 상태 동기화
- 재생/종료 버튼 클릭 시 이벤트 버블링 방지를 위해 e.stopPropagation 추가

faber-h commented on src/hooks/useChannelPlayback.jsx:
```diff
@@ -32,16 +32,28 @@ const useChannelPlayback = (mode) => {
   };
 
-  const handlePlayPauseOnce = async () => {
+  const handlePlayPauseOnce = async (e) => {
```
중복 실행 방지를 좀 더 명확하게 드러내기 위해 `handlePlayPauseOnce`를 `safeTogglePlayback` 같은 함수명은 어떨까요?
두 분의 의견이 궁금합니다!
@Eun0713 @devjaeyoon

d0-neii replied:
저도 `handlePlayPauseOnce`보다 `safeTogglePlayback`처럼 역할을 더 명확하게 드러내는 이름이 좋다고 생각합니다!
중복 실행 방지라는 함수의 의도가 더 직관적으로 전달되는 것 같아요~

devjaeyoon replied:
두 분 모두 좋은 의견 감사드립니다. 저는 `safeTogglePlayback`이라는 함수명이 `TogglePlayback`을 지켜준다라는 느낌을 받아서 해당 함수는 `handlePlayPause`를 호출하며 중복 호출을 방지하기 위해 `isProcessing`을 넣어준 건데 지켜준다와는 거리가 멀고 추상적이라고 생각했어요. 그러면 guarded 함수임을 표현하는 `guardedPlayPause`는 어떤가요? @Eun0713 @faber-h

faber-h replied:
`guardedPlayPause`도 지켜준다는 느낌의 네이밍인 것 같아요.
조금 더 구체적으로 중복 클릭 방지의 느낌이 드러나는 이름으로 `handleClickWithLock`은 어떠신가요?
클릭 이벤트 핸들러임도 명확하게 드러나서 사용 맥락이 더 분명하게 느껴지는 장점이 있는 것 같아요.
@Eun0713 @devjaeyoon

devjaeyoon replied:
`handleClickWithLock`은 잠금과 함께 클릭을 다룬다는 느낌으로 중복 호출을 방지하는거지 잠근다와는 거리가 약간 멀다 생각해요. 그러면 안전하게 handlePlayPause를 다룬다는 느낌으로 `handlePlayPauseSafely`는 어떤가요?

d0-neii replied:
`handlePlayPauseSafely`도 함수의 목적이 잘 드러나는 이름이라고 생각해요.
저도 그런 맥락에서, 중복 실행 자체를 더 직접적으로 표현한 `preventDoublePlayback`같은 이름도 좋을 것 같아요!
함수 이름만 보고도 중복 재생을 막는 역할이라는 걸 직관적으로 이해할 수 있을 거 같아서 함께 고려해보면 어떨까요??

faber-h replied:
`Once`, `Double` 같은 키워드는 약간 횟수의 제한으로 느껴질 수 있어서,
중복 실행 자체를 더 명확히 드러내는 `preventDuplicateClick` 같은 이름도 함께 고려해보면 어떨까요?
클릭 중복 방지라는 함수의 의도가 이름에서 바로 전달되는 것 같아요.

devjaeyoon replied:
`preventDuplicateClick` 좋은 것 같아요! @Eun0713 다은님도 괜찮으신가요?

d0-neii replied:
네 저도 `preventDuplicateClick` 좋은 것 같습니다! 의미가 명확히 들어나는 것 같아요~

devjaeyoon replied:
그러면 `preventDuplicateClick`로 하겠습니다. 두 분 모두 늦은 시간까지 감사드립니다!

devjaeyoon committed:
refactor: 중복 클릭 방지 목적의 함수명을 preventDuplicateClick으로 변경

함수의 역할이 handlePlayPause를 안전하게 호출하는 것임을 고려하여
handlePlayPauseOnce에서 preventDuplicateClick으로 이름을 변경함.
이로써 함수 목적을 더 명확하게 전달할 수 있도록 개선.

faber-h reviewed (approved):
LGTM!

Eun0713 reviewed (approved):
LGTM✨
